### PR TITLE
Fixed containerChildrenMutated method so it better checks whether there were any children mutations

### LINKED ISSUE #117: Word is removed when spellchecker is used
**Steps to reproduce:**

1. Open `article.html` manual test (http://localhost:8125/ckeditor5-presets/tests/manual/article.html).
2. Remove all content and paste **B das daold** _Itad aslic_.
3. Use spellchecker to fix "daold" word.

**Expected result:**

Word is fixed.

**Current result:**

Word is removed.

**Error:**

```
Uncaught TypeError: Cannot read property 'index' of undefined
    at MutationHandler._handleTextNodeInsertion (article.js:51303)
    at MutationHandler.handle (article.js:51161)
    at Input._handleMutations (article.js:51112)
    at Document.listenTo (article.js:51057)
    at Document.fire (article.js:4309)
    at MutationObserver._onMutations (article.js:26922)
```

**GIF**

aug-21-2017 16-56-16

### PR DESCRIPTION
**Pull request by scofalik**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: Fixed an error where using spellchecker on a word with style applied sometimes resulted in that word being removed. Closes #117.

### CODE REVIEW

File: src/input.js [hunk 1]
```diff
@@ -423,12 +423,12 @@ function containerChildrenMutated( mutations ) {
 
 	// Check if all mutations are `children` type, and there is no single text node mutation.
```
Comment (Reinmar) [inline src/input.js]: This comment needs to be synced with the code.

Comment (Reinmar) [reply src/input.js]: And the function description could be too.

**Comment by Reinmar** [comment]
I understand this change as

* previously: "apply this algorithm when none of the mutations was a text or single text child mutation",
* now: "apply this algorithm when at least one mutation is not text or single text child mutation".

Which seems to make a perfect sense. But please confirm it and improve the comments. Also, I'd love if you stretched your imagination and wrote more tests – especially for cases which should not be caught by this algorithm.

Finally, I started wondering whether it's tested what happens if mutations change more than a content of one model element.

File: src/input.js (hunk 1 above)
Comment (scofalik) [reply src/input.js]: Function description is fine, but I re-arranged it. The comment will be changed.

**Comment by scofalik** [comment]
> But please confirm it
I confirm. It was even described like this in the comment which helped me realize that the implementation was wrong.

**Comment by scofalik** [comment]
> Finally, I started wondering whether it's tested what happens if mutations change more than a content of one model element.

We have various tests already:
* `should handle mutations switching inner and outer node when adding new text node before` tests more than one children mutation;
* `should do nothing if mutations does not have common ancestor` tests what happens when mutation takes place in a "weird" node
* and I've now added `should handle multiple text mutations` to handle a mutation with two text changes, though I don't know what real case could create such mutations.

> Also, I'd love if you stretched your imagination and wrote more tests – especially for cases which should not be caught by this algorithm.

Honestly, I've wrote a test case for this exact change and other tests are for "cases which should not be caught by this algorithm". These are cases when there are only text mutations or there is a children mutation of one inserted element (like `should handle text node insertion` test). If you have any other scenario on mind let me know and I'll write a test for it.

**Comment by scofalik** [comment]
I mean, take a look here:
```javascript
handle( mutations, viewSelection ) {
	if ( containerChildrenMutated( mutations ) ) {
		this._handleContainerChildrenMutations( mutations, viewSelection );
	} else {
		for ( const mutation of mutations ) {
			// Fortunately it will never be both.
			this._handleTextMutation( mutation, viewSelection );
			this._handleTextNodeInsertion( mutation );
		}
	}
}
```
The bug was caused because `containerChildrenMutated` incorrectly recognized a scenario that happened. This was because of incorrect implementation in that function. No logic changed in algorithms that actually did anything.

**Pull request merged**